cloudstack/data_source_cloudstack_common_schema.go

```diff
@@ -0,0 +1,25 @@
+package cloudstack
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceFiltersSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet,
+		Optional: true,
```

Comment:
Why make this optional? I assume this call is useless without a filter right? So I suggest making this required instead.